lib/charms/opensearch/v0/opensearch_peer_clusters.py

```diff
@@ -590,3 +594,61 @@ def _deployment_type(config: PeerClusterConfig, start_mode: StartMode) -> Deploy
             if not config.init_hold
             else DeploymentType.FAILOVER_ORCHESTRATOR
         )
+
+    def rel_data_from_redacted_str(self, redacted_dict_str: str) -> PeerClusterRelData | None:
```

Comment:
i don't think this method can return `None`, a renaming may also make sense here:
```suggestion
    def rel_data_from_str(self, redacted_dict_str: str) -> PeerClusterRelData:
```